sdk/src/main/java/io/radar/sdk/util/RadarLogBuffer.kt

```diff
@@ -0,0 +1,199 @@
+package io.radar.sdk.util
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import io.radar.sdk.Radar
+import io.radar.sdk.model.RadarLog
+import org.json.JSONObject
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.util.*
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.ReentrantLock
+
+/**
+ * Wraps the filesystem to support log caching
+ */
+internal class RadarLogBuffer(
+    private val context: Context,
+    private val executor: ExecutorService = Executors.newSingleThreadExecutor()
+) {
+
+    private companion object {
+        const val DIRECTORY = "RadarLogs"
+
+        /**
+         * Maximum number of logs in the buffer.
+         */
+        const val LOG_BUFFER_SIZE_LIMIT = 500
+    }
+
+    /**
+     * Lock for handling changes to [currentFile]
+     */
+    private val fileLock = ReentrantLock()
+
+    /**
+     * Lock for handling changes to the files within [DIRECTORY]
+     */
+    private val directoryLock = ReentrantLock()
+
+    /**
+     * Contains the name of the current log file
+     */
+    private var currentFile: String
+
+    /**
+     * Number of log entries in the current log file.
+     */
+    val size: AtomicInteger
```

Comment:
I need to read the locking logic, but do we need an AtomicInteger if we have the locks above?

Reply:
it could likely be removed if the `isFull` function uses the fileOp lock